File: frontend/src/services/chat.service.js
import api from './api';

export const chatService = {
  createSession: async (subjectId) => {
    const response = await api.post('/chat/sessions', { subjectId });
    return response.data.data;
  },

  getSessions: async () => {
    const response = await api.get('/chat/sessions');
    return response.data.data;
  },

  getSessionHistory: async (sessionId) => {
    const response = await api.get(`/chat/sessions/${sessionId}/messages`);
    return response.data.data;
  },

  sendMessage: async (sessionId, question, approach = 'RAG') => {
    const response = await api.post(`/chat/sessions/${sessionId}/messages`, { sessionId, question, approach });
    return response.data.data;
  },

  rateMessage: async (messageId, rating, feedbackType = 'NONE') => {
    const response = await api.put(`/chat/sessions/messages/${messageId}/rating`, { rating, feedbackType });
    return response.data.data;
  }
};

File: services/core_backend/src/main/java/com/edu/chatbot/chat/service/ChatService.java
package com.edu.chatbot.chat.service;

import com.edu.chatbot.chat.dto.request.ChatRequest;
import com.edu.chatbot.chat.dto.response.ChatResponseDTO;
import com.edu.chatbot.chat.dto.response.MessageDTO;
import com.edu.chatbot.chat.dto.response.SessionDTO;

import java.util.List;

public interface ChatService {
    SessionDTO createSession(Long subjectId, Long userId);
    List<SessionDTO> getUserSessions(Long userId);
    List<MessageDTO> getSessionHistory(Long sessionId, Long userId);

    ChatResponseDTO processRAGMessage(ChatRequest request, Long userId);
    ChatResponseDTO processFineTuneMessage(ChatRequest request, Long userId);
    ChatResponseDTO dispatch(ChatRequest request, Long userId);
    void rateMessage(Long messageId, Integer rating, String feedbackType);
}
